flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java

```diff
@@ -421,6 +421,8 @@ public void testCancellationOfCanceledTerminalDoesNotThrowException() throws Exc
 
         // wait for job to finish
         dispatcherGateway.requestJobResult(jobId, TIMEOUT).get();
+        dispatcher.getJobTerminationFuture(jobId, TIMEOUT).get();
```

Comment:
It took me a bit to figure out why we need this change. It's related to the `thenComposeAsync` which you added in `Dispatcher.runJob` to run the `Dispatcher#removeJob` method in the main thread. I commented on that one separatly. But we could revert this change if we agree that it's not necessary.